### PR TITLE
Add ability to upgrade to latest version in rustic-outdated-mode

### PR DESCRIPTION
**Pull request by psibi**
Implements two features:

- Ability to mark a specific crate to upgrade it to the latest available version
- Ability to mark all creates to upgrade to the latest available version

The README has also been updated with the relevant documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -500,8 +500,11 @@ can use most commands you know from the emacs package menu. This
 option requires the rust package `cargo-outdated` to be installed
 before being used.
 
-- `u` mark single crate for upgrade
-- `U` mark all upgradable crates
+- `u` mark single crate for upgrade. Asks using minibuffer the exact
+  version that should be used for upgrade.
```
Comment (brotzeit) [inline README.md:504]: "mark single crate for upgrade. Asks using minibuffer the exact
  version that should be used for upgrade"
-> 
"mark single crate for upgrade and prompt user for version"

File: README.md [hunk 2]
```diff
@@ -500,8 +500,11 @@ can use most commands you know from the emacs package menu. This
...
+- `U` mark all upgradable crates.
+- `l` mark single crate for upgrading to latest available package
```
Comment (brotzeit) [inline README.md:506]: latest available package -> latest version

File: README.md [hunk 3]
```diff
@@ -500,8 +500,11 @@ can use most commands you know from the emacs package menu. This
...
+- `L` mark all crates to latest available package
```
Comment (brotzeit) [inline README.md:507]: I would also use version instead of available package here.

File: rustic-cargo.el [hunk 4]
```diff
@@ -349,6 +351,53 @@ Execute process in PATH."
...
+;;;###autoload
+(defun rustic-cargo-mark-latest-upgrade ()
+  "Mark an upgradable package to the latest available version."
+  (interactive)
+  (let* ((crate (tabulated-list-get-entry (point)))
+         (v (substring-no-properties (elt crate 3)))
+         (inhibit-read-only t))
+    (when v
+      (save-excursion
+        (goto-char (line-beginning-position))
```
Comment (brotzeit) [inline rustic-cargo.el:363]: You could add a variable `line-beg`.

File: rustic-cargo.el [hunk 5]
```diff
@@ -349,6 +351,53 @@ Execute process in PATH."
...
+        (save-match-data
+          (when (search-forward (elt crate 0))
+            (replace-match (propertize (elt crate 0)
+                                       'font-lock-face
+                                       'rustic-cargo-outdated-upgrade)))
```
Comment (brotzeit) [inline rustic-cargo.el:368]: Maybe use a `lambda` ?

Comment (psibi) [reply rustic-cargo.el:368]: Sorry, not sure I understand what to do here. Can you probably provide a sample code ? Thanks!

File: README.md (hunk 2 above)
Comment (psibi) [reply README.md]: Thanks, fixed.

File: README.md (hunk 1 above)
Comment (psibi) [reply README.md]: Fixed.

File: rustic-cargo.el (hunk 4 above)
Comment (psibi) [reply rustic-cargo.el]: Fixed, thanks!

File: README.md (hunk 3 above)
Comment (psibi) [reply README.md]: Fixed, thanks!

File: rustic-cargo.el (hunk 5 above)
Comment (brotzeit) [reply rustic-cargo.el:369]: I mean you could put the call to `replace-match` into a lambda that is in the let bindings like in https://github.com/brotzeit/rustic/blob/01c3b34016656d7dcf36511b9f6410d71a112e29/rustic-racer.el#L325

Comment (psibi) [reply rustic-cargo.el:369]: Thanks, updated.

**Comment by brotzeit** [comment]
Thanks!

**Pull request merged**